desktop/src/main/java/org/ergoplatform/desktop/wallet/CreateWalletComponent.kt

```diff
@@ -0,0 +1,45 @@
+package org.ergoplatform.desktop.wallet
+
+import androidx.compose.material.ScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import com.arkivanov.decompose.ComponentContext
+import com.arkivanov.decompose.router.pop
+import com.arkivanov.decompose.router.push
+import org.ergoplatform.appkit.Mnemonic
+import org.ergoplatform.appkit.SecretString
+import org.ergoplatform.desktop.ui.navigation.NavClientScreenComponent
+import org.ergoplatform.desktop.ui.navigation.NavHostComponent
+import org.ergoplatform.desktop.ui.navigation.ScreenConfig
+import org.ergoplatform.desktop.ui.showSensitiveDataCopyDialog
+
+class CreateWalletComponent(
+    private val componentContext: ComponentContext,
+    private val navHost: NavHostComponent
+) : NavClientScreenComponent(navHost), ComponentContext by componentContext {
+
+    override val appBarLabel: String
+        get() = ""
+
+    override val fullScreen: Boolean
+        get() = true
+
+    @Composable
+    override fun renderScreenContents(scaffoldState: ScaffoldState?) {
+        CreateWalletScreen(
+            onBack = {
+                router.pop()
+                mnemonic.erase()
+            },
+            onProceed = {
+                router.push(ScreenConfig.ConfirmCreateWallet(mnemonic))
+            },
+            onCopy = {
+                showSensitiveDataCopyDialog(navHost, mnemonic.toStringUnsecure())
+            },
+            remember { mnemonic.toStringUnsecure() }
```

Comment:
It is to show the mnemonic on screen to the user. If it would not happen, the user could not note down the mnemonic. :-)

By the way, as we deal with compose in the future more often, on Android as well, this page is very important to understand. It explains what the `remember` is doing here and that it does not lead to a longer lifetime of the unsecure mnemonic, but instead to less calls to the method: https://developer.android.com/jetpack/compose/state